src/wallet/api/wallet.cpp

```diff
@@ -2016,7 +2036,8 @@ void WalletImpl::doRefresh()
         }
     } catch (const std::exception &e) {
         setStatusError(e.what());
-    }
+    }while(!rescan && (rescan=m_refreshShouldRescan.exchange(false))); // repeat if not rescanned and rescan was requested
```

Comment:
This seems odd. It's basically a test for "did we just miss it by not much, and if so do it now anyway", right ?

Also, it should probably not loop if the catch just above triggered

Reply:
> This seems odd. It's basically a test for "did we just miss it by not much, and if so do it now anyway", right ?

Well, I could explain this logic as "If someone requested to rescan, and we're refreshing, let's refresh again (with rescan)".

> Also, it should probably not loop if the catch just above triggered
Agree, will update this PR

Reply:
That's what I thought. It seems a bit weird since it'll get picked up next time refresh is called without the extra loop, but I suppose it's OK.